lnrpc/routerrpc/router.proto

```diff
@@ -490,6 +490,67 @@ message BuildRouteResponse {
     lnrpc.Route route = 1;
 }
 
+message CircuitKey {    
+    /// The id of the channel that the htlc arrived at our node on.
+    uint64 incoming_chan_id = 1;
+
+    /// The index of the incoming htlc in the incoming channel.
+    uint64 htlc_id = 2;
+}
+
+message ForwardHtlcInterceptRequest {
```

Comment:
It'd be great to include the sphinx packet in this message. It would allow a receiver to check amount and expiry as it would if the HTLC had been forwarded through a link. This is useful for reverse swaps.

Reply:
@champo thanks for the feedback.
Currently the fields `amount_sat` `expiry` are what you are looking for as they are the actual values that would have been used in the forwarded packet in the default switch behavior.

Reply:
A third-party receiver (say a mobile app) shouldn't trust the fields. They would want to check the information against the sphinx. It would also enable usage of new fields like payment secret which is only known to the receiver.

Reply:
@champo I agree. At first I exposed only the fields that are required for the use cases I mentioned in the PR description. I do think it makes sense to expose all the forwarding info (such as outgoing channel, tlv fields and raw sphinx packet) as other use cases may need this info in order to take a decision at interception time.

Reply:
@Roasbeef in our case, the end node and the interceptor communicate over HTTP to negotiate the settling of the HTLC. The end node can then decode the sphinx packet to reduce trust.